dart/simulation/World.h

```diff
@@ -107,57 +126,47 @@ class World : public virtual common::Subject
   //--------------------------------------------------------------------------
 
   /// Get the indexed skeleton
-  dynamics::Skeleton* getSkeleton(size_t _index) const;
+  dynamics::SkeletonPtr getSkeleton(size_t _index) const;
 
   /// Find a Skeleton by name
   /// \param[in] The name of the Skeleton you are looking for.
-  /// \return If the skeleton does not exist then return NULL.
-  dynamics::Skeleton* getSkeleton(const std::string& _name) const;
+  /// \return If the skeleton does not exist then return nullptr.
+  dynamics::SkeletonPtr getSkeleton(const std::string& _name) const;
 
   /// Get the number of skeletons
   size_t getNumSkeletons() const;
 
   /// Add a skeleton to this world
-  std::string addSkeleton(dynamics::Skeleton* _skeleton);
-
-  /// Remove a skeleton from this world without deleting it
-  void withdrawSkeleton(dynamics::Skeleton* _skeleton);
+  std::string addSkeleton(dynamics::SkeletonPtr _skeleton);
 
-  /// Remove a skeleton in this world and delete it
-  void removeSkeleton(dynamics::Skeleton* _skeleton);
+  /// Remove a skeleton from this world
+  void removeSkeleton(dynamics::SkeletonPtr _skeleton);
 
-  /// Remove all skeletons from this world, and get pointers to them
-  std::set<dynamics::Skeleton*> withdrawAllSkeletons();
-
-  /// Remove all the skeletons in this world and delete them
-  void removeAllSkeletons();
+  /// Remove all the skeletons in this world, and return a set of shared
+  /// pointers to them, in case you want to recycle them
+  std::set<dynamics::SkeletonPtr> removeAllSkeletons();
 
   /// Get the dof index for the indexed skeleton
   int getIndex(int _index) const;
 
   /// Get the indexed Entity
-  dynamics::Entity* getEntity(size_t _index) const;
+  dynamics::SimpleFramePtr getFrame(size_t _index) const;
 
   /// Find an Entity by name
-  dynamics::Entity* getEntity(const std::string& _name) const;
+  dynamics::SimpleFramePtr getFrame(const std::string& _name) const;
 
   /// Get the number of Entities
-  size_t getNumEntities() const;
+  size_t getNumFrames() const;
 
   /// Add an Entity to this world
-  std::string addEntity(dynamics::Entity* _entity);
-
-  /// Remove an Entity from this world without deleting it
-  void withdrawEntity(dynamics::Entity* _entity);
+  std::string addFrame(dynamics::SimpleFramePtr _frame);
```

Comment:
How about `addSimpleFrame()`? `addFrame()` sounds like it takes `Frame` or any classes inherited `Frame`.